README.md

```diff
@@ -1,12 +1,75 @@
 celvet
 ======
 
+**celvet** is a multitool for linting CRDs (`CustomResourceDefinition`) in
+Kubernetes. `celvet` also provides particular insight into [validation rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules),
+checking that they do not exceed the resource constraints provided by
+Kubernetes' API server. The tool is deliberately made to be useful both in an
+ad-hoc fashion by developers in addition to being part of a CI/CD pipeline, so
+`celvet` supports output in JSON in addition to machine-readable text.
+
+What does celvet catch?
+-----------------------
+
+* **CRD parsing errors** - celvet will early exit if it finds that a CRD
+    contains invalid YAML/JSON.
+* **Missing limits on variable-length fields** - `maxLength`, `maxItems`, etc.
+* **Validation rule parsing errors**  - if celvet has issues parsing a
+    validation rule, it will output the errors.
+* **Exceeding resource constraints for validation rules** - if a validation
+    rule exceeds the per-rule cost budget, celvet will output an error.
+    Additionally, if the sum total of rules on the CRD exceed the per-CRD cost
+    budget, celvet will output an error there as well.
+
+Installing
+----------
+
+celvet can be installed with `go get`:
+
+    go get github.com/DangerOnTheRanger/celvet
+
 Usage
 -----
 
 ```
 celvet crd-file
 ```
 
-If `celvet` finds any linting errors, it will print them to stdout and return
-a non-zero error code.  
\ No newline at end of file
+If celvet finds any linting errors, it will print them to stdout and return
+a non-zero error code. For example, given one of the example CRDs from the [k8s documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition),
+celvet will output:
+```
+string "<root>.spec.cronSpec" missing maxLength
+string "<root>.spec.image" missing maxLength
+```
+
+The `-o`/`--output` option can be used to change the output format to JSON if
+`json` is passed to the flag; `text` is the default option but can be
+explicitly passed. The same file but with `-o json` passed will give the
+following (the data has been pretty-printed here for readability):
+```json
+{
+  "limitErrors": [
+    {
+      "path": "<root>.spec.cronSpec",
+      "type": 2
+    },
+    {
+      "path": "<root>.spec.image",
+      "type": 2
+    }
+  ],
+  "costErrors": null,
+  "compileErrors": null,
+  "otherErrors": []
```

Comment:
Why are the 1st two null and the last an empty array?

Reply:
This was using an old commit that didn't include https://github.com/DangerOnTheRanger/celvet/pull/8/commits/2cf6fd9e410572d7d4c2c0ffa15bd0162a03bf9e - I've re-linted the same file from that branch, so the output should be cleaner now.